prow/cmd/deck/main.go

```diff
@@ -1172,18 +1187,28 @@ func handleRerun(prowJobClient prowv1.ProwJobInterface, createProwJob bool) http
 		// On Prow instances that require auth even for viewing Deck this is okayish, because the Prowjob UUID
 		// is hard to guess
 		// Ref: https://github.com/kubernetes/test-infra/pull/12827#issuecomment-502850414
-		if createProwJob {
-			if r.Method != http.MethodPost {
-				http.Error(w, "request must be of type POST", http.StatusMethodNotAllowed)
-				return
-			}
-			if _, err := prowJobClient.Create(&newPJ); err != nil {
-				logrus.WithError(err).Error("Error creating job")
-				http.Error(w, fmt.Sprintf("Error creating job: %v", err), http.StatusInternalServerError)
-				return
+		if r.Method == http.MethodPost {
```

Comment:
Why is Put, Get, etc valid?

Reply:
We want the rerun endpoint to still work for `kubectl create -f "https://prow.k8s.io/rerun?prowjob=blahblah"` (which is a Get request), even when rerunning jobs directly is enabled. This check is just a way of determining whether we want to trigger the job or write the YAML to the page.

Reply:
Might be worth moving this into different functions

Reply:
I'm not sure that would make things cleaner, since the function to trigger the rerun needs a ton of parameters (`r`, `w`, `cfg`, `newPJ`, `prowJobClient`, ...)

Reply:
Doesn't seem like we need that many for the get scenario.

Reply:
You're right, factored it out